pkg/cfn/manager/cluster.go

```diff
@@ -58,22 +58,34 @@ func (c *StackCollection) DescribeClusterStack() (*Stack, error) {
 }
 
 // RefreshFargatePodExecutionRoleARN reads the CloudFormation stacks and
-// their output values, and set the Fargate pod execution role ARN to
+// their output values, and sets the Fargate pod execution role ARN to
 // the ClusterConfig.
 func (c *StackCollection) RefreshFargatePodExecutionRoleARN() error {
+	fargateOutputs := map[string]outputs.Collector{
+		outputs.FargatePodExecutionRoleARN: func(v string) error {
+			c.spec.IAM.FargatePodExecutionRoleARN = &v
+			return nil
+		},
+	}
 	stack, err := c.DescribeClusterStack()
 	if err != nil {
 		return err
 	}
-	return outputs.Collect(*stack,
-		map[string]outputs.Collector{
-			outputs.FargatePodExecutionRoleARN: func(v string) error {
-				c.spec.IAM.FargatePodExecutionRoleARN = &v
-				return nil
-			},
-		},
-		nil,
-	)
+	if err := outputs.Collect(*stack, nil, fargateOutputs); err != nil {
+		return err
+	}
+
+	if c.spec.IAM.FargatePodExecutionRoleARN == nil {
+		if err := c.FixClusterCompatibility(); err != nil {
+			return errors.Wrap(err, "error fixing cluster compatibility")
```

Comment:
Should we be more specific to the user here? Explain what was that eksctl tried to achieve?

Reply:
Perhaps we can add the message inside FixClusterCompatibility and here only return the error?

Reply:
Adding the message inside FixClusterCompatibility would require duplicating the error message for all error return paths.
I'm thinking of adding a log message before the call to FixClusterCompatibility:
```golang
logger.Info("Fargate pod execution role is missing, fixing cluster stack to add Fargate resources")
```
WDYT?